NPE with nested null when creating a UDT value in SQL

The following test:
```java
    @Test
    public void test2() {
        instance().getSql().execute("CREATE TYPE Office ("
                + "id BIGINT, "
                + "name VARCHAR "
                + ") OPTIONS ('format'='compact', 'compactTypeName'='OfficeCompactType')");

        instance().getSql().execute("CREATE TYPE Organization ("
                + "id BIGINT, "
                + "name VARCHAR, "
                + "office Office"
                + ") OPTIONS ('format'='compact', 'compactTypeName'='OrganizationCompactType')");

        instance().getSql().execute(
                "CREATE MAPPING test ("
                        + "__key BIGINT,"
                        + "organization Organization"
                        + ")"
                        + "TYPE IMap "
                        + "OPTIONS ("
                        + "'keyFormat'='bigint',"
                        + "'valueFormat'='compact',"
                        + "'valueCompactTypeName'='UserCompactType'"
                        + ")");

        instance().getSql().execute("insert into test values (1, (2, 'orgName', null))");
    }
```
Fails with:
```
com.hazelcast.sql.HazelcastSqlException: Cannot invoke "com.hazelcast.sql.impl.expression.RowValue.getValues()" because "rowValue" is null

	at com.hazelcast.sql.impl.QueryUtils.toPublicException(QueryUtils.java:86)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:224)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:175)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:171)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:167)
	at com.hazelcast.sql.SqlService.execute(SqlService.java:89)
	at com.hazelcast.jet.sql.impl.type.UdtObjectToJsonFunctionTest.test2(UdtObjectToJsonFunctionTest.java:153)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: Cannot invoke "com.hazelcast.sql.impl.expression.RowValue.getValues()" because "rowValue" is null
	at com.hazelcast.jet.sql.impl.inject.UpsertTargetUtils.setCompactFields(UpsertTargetUtils.java:283)
	at com.hazelcast.jet.sql.impl.inject.UpsertTargetUtils.setCompactFields(UpsertTargetUtils.java:327)
	at com.hazelcast.jet.sql.impl.inject.UpsertTargetUtils.convertRowToCompactType(UpsertTargetUtils.java:116)
	at com.hazelcast.jet.sql.impl.inject.CompactUpsertTarget.lambda$createRowValueInjector$14(CompactUpsertTarget.java:112)
	at com.hazelcast.jet.sql.impl.connector.keyvalue.KvProjector.project(KvProjector.java:92)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.Collections$2.tryAdvance(Collections.java:4853)
	at java.base/java.util.Collections$2.forEachRemaining(Collections.java:4861)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at com.hazelcast.jet.sql.impl.opt.physical.InsertMapPhysicalRel.lambda$entriesFn$0(InsertMapPhysicalRel.java:85)
	at com.hazelcast.jet.sql.impl.PlanExecutor.execute(PlanExecutor.java:428)
	at com.hazelcast.jet.sql.impl.SqlPlanImpl$IMapInsertPlan.execute(SqlPlanImpl.java:1127)
	at com.hazelcast.sql.impl.SqlServiceImpl.query0(SqlServiceImpl.java:269)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:207)
	... 17 more
```

--- a/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/UpsertTargetUtils.java
+++ b/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/UpsertTargetUtils.java
@@ -322,10 +322,14 @@ public final class UpsertTargetUtils {
                     recordBuilder.setTimestampWithTimezone(field.getName(), (OffsetDateTime) fieldValue);
                     break;
                 case OBJECT:
-                    final GenericRecordBuilder nestedRecordBuilder = GenericRecordBuilder
-                            .compact(field.getDataType().getObjectTypeMetadata());
-                    setCompactFields((RowValue) fieldValue, field.getDataType(), nestedRecordBuilder);
-                    recordBuilder.setGenericRecord(field.getName(), nestedRecordBuilder.build());
+                    if (fieldValue == null) {
+                        recordBuilder.setGenericRecord(field.getName(), null);
+                    } else {
+                        final GenericRecordBuilder nestedRecordBuilder = GenericRecordBuilder
+                                .compact(field.getDataType().getObjectTypeMetadata());
+                        setCompactFields((RowValue) fieldValue, field.getDataType(), nestedRecordBuilder);
+                        recordBuilder.setGenericRecord(field.getName(), nestedRecordBuilder.build());
+                    }
                     break;
                 case INTERVAL_YEAR_MONTH:
                 case INTERVAL_DAY_SECOND:

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/type/BasicNestedFieldsTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/type/BasicNestedFieldsTest.java
@@ -515,6 +515,35 @@ public class BasicNestedFieldsTest extends SqlTestSupport {
                 .hasMessage("From line 1, column 27 to line 1, column 66: Comparison operators are not supported for ROW type");
     }
 
+    @Test
+    public void test_nullValueInRow() {
+        instance().getSql().execute("CREATE TYPE Office ("
+                + "id BIGINT, "
+                + "name VARCHAR "
+                + ") OPTIONS ('format'='compact', 'compactTypeName'='OfficeCompactType')");
+
+        instance().getSql().execute("CREATE TYPE Organization ("
+                + "id BIGINT, "
+                + "name VARCHAR, "
+                + "office Office"
+                + ") OPTIONS ('format'='compact', 'compactTypeName'='OrganizationCompactType')");
+
+        instance().getSql().execute(
+                "CREATE MAPPING test ("
+                        + "__key BIGINT,"
+                        + "organization Organization"
+                        + ")"
+                        + "TYPE IMap "
+                        + "OPTIONS ("
+                        + "'keyFormat'='bigint',"
+                        + "'valueFormat'='compact',"
+                        + "'valueCompactTypeName'='UserCompactType'"
+                        + ")");
+
+        instance().getSql().execute("INSERT INTO test VALUES (1, (2, 'orgName', null))");
+        assertRowsAnyOrder("SELECT (organization).office FROM test", rows(1, new Object[]{null}));
+    }
+
     private User initDefault() {
         testInstance().getSql().execute(format("CREATE TYPE UserType (id BIGINT, name VARCHAR, organization OrganizationType) "
                 + "OPTIONS ('format'='java', 'javaClass'='%s')", User.class.getName()));

